### PR TITLE
fix num_tasks in spark integration

### PR DESCRIPTION
**Pull request by jmoralez**
Moves the logic of the distributed processing to separate methods to make it easier to follow the original methods (fit and predict).

During this refactor I found and fixed the error here: https://github.com/Nixtla/neuralforecast/blob/0e43ae0e7bc42294be95e1f4ebb497d7de1817d7/neuralforecast/common/_base_model.py#L333

After this I reran and updated the sample notebook, which was missing from the sidebar, so this adds it as well.

### CODE REVIEW

File: nbs/common.base_model.ipynb
Comment (elephaint) [inline nbs/common.base_model.ipynb:1]: I think the change is fine, but isn't it easier to just up the Pytorch requirement for neuralforecast going forward?

File: nbs/core.ipynb
Comment (elephaint) [inline nbs/core.ipynb:1]: I understand we only support Spark dataframes here - do we intend to extend it to other types of distributed dfs in the future or do we want to write separate functions for that?

**Review by elephaint — APPROVED** [review]

File: nbs/common.base_model.ipynb
Comment (jmoralez) [reply nbs/common.base_model.ipynb:1]: It's not worth it dropping support for an entire minor version (2.0) for just one small change.

File: nbs/core.ipynb
Comment (jmoralez) [reply nbs/core.ipynb:1]: Each framework has its own way of performing DDP, see for example [dask](https://github.com/saturncloud/dask-pytorch-ddp) and [ray](https://docs.ray.io/en/latest/train/getting-started-pytorch-lightning.html). So we may be able to reuse the dataset but we'll have to define different training functions for each.

**Pull request merged**